include/openenclave/internal/lockless_queue.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _LOCKLESS_QUEUE_H_
+#define _LOCKLESS_QUEUE_H_
+
+#if _MSC_VER
+#include <Windows.h>
```

Comment:
If the queue is used within the enclave, it will not have access to system headers. So Windows.h may not be available.